components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/CustomTabWindowFeature.kt

```diff
@@ -76,17 +76,36 @@ class CustomTabWindowFeature(
                     val windowRequest = state.content.windowRequest
                     if (windowRequest?.type == WindowRequest.Type.OPEN) {
                         val intent = configToIntent(state.config)
-                        try {
-                            intent.launchUrl(activity, windowRequest.url.toUri())
-                        } catch (e: ActivityNotFoundException) {
-                            crashReporter?.submitCaughtException(e)
+                        val url = windowRequest.url.toUri()
+                        // Setting url for intent in order to check if we are able to open that url
+                        intent.intent.data = url
+                        if (canHandleIntent(intent)) {
+                            intent.launchUrl(activity, url)
+                        } else {
+                            // We could only get here if the above intent is for our application
+                            // and we are not registered to handle its schemes.
+                            // Let's log this to better asses how often this happens in real world and
+                            // if we need to add new schemes to properly support this workflow.
+                            // See Fenix #8412
+                            try {
+                                intent.launchUrl(activity, url)
```

Comment:
Looks like both `canHandleIntent(activity, url)` true and false case both calls `intent.launchUrl()`.  Why do the extra check?

Reply:
When `canHandleIntent` returns:
- true: 
We know that the client can open the link and we simply delegate it to do so.
- false:
We know that the client can't handle the link but we try to give to it a stacktrace with even more information about why the call to open the link failed.
In this case the `intent.launchUrl(activity, url)` it's clear that it will not actually open the url, but just throw an exception which we'll forward to the client.
To give a bit more context about this I wrote the comment above the `try-catch`.

Reply:
I see.  But couldn't we (like the change I landed) just call `launchUrl` in the try catch and handle the error?

Reply:
I just found it to be more clear and I already had the patch and tests for that scenario. I think the difference is minor.

Reply:
Agreed.  I would preferred we go straight to try catch without the extra check.  Thanks,

Reply:
Hello @rocketsroger, I agree with you that we should only keep the `try catch` without the extra check and I updated the PR accordingly. Thanks!